core/src/event_id.rs

```diff
@@ -28,35 +36,73 @@ impl EventId {
     ///
     /// This ensures that identical event names always produce the same event ID, while
     /// providing good distribution properties to minimize collisions between different names.
-    pub fn from_name(name: impl AsRef<str>) -> Self {
-        let digest_word = hash_string_to_word(name.as_ref());
+    ///
+    /// The event name is stored and will be displayed in error messages.
+    pub fn from_name(name: impl Into<String>) -> Self {
```

Comment:
Looking at `FastProcessor::op_emit`, the name is never present on `EventId`s passed to the host's `on_event` function, right? Is there a way to change this or make it otherwise possible for miden VM users to lookup the name from an ID? This would be just for debugging purposes or error messages, so in branches where performance is not critical.